model/event_context_test.go

```diff
@@ -0,0 +1,368 @@
+	return -1
+}
+func (f *eventContextFixture) GetConclusion() string {
+	return f.conclusion
+}
+func (f *eventContextFixture) GetStatus() string {
+	return f.status
+}
+func (f *eventContextFixture) GetRepository() string {
+	return f.repository
+}
+func (f *eventContextFixture) GetName() string {
+	return f.name
+}
+func (f *eventContextFixture) GetInstallationID() int64 {
+	return -1
+}
+func (f *eventContextFixture) GetCommitHash() string {
+	return f.commitHash
+}
+func (f *eventContextFixture) Log() {
+}
+
+func createPipelineConfiguration() []config.PipelineConfig {
+	return []config.PipelineConfig{
+		{
+			Organization: "a",
+			Repository:   "b",
+			Workflow:     "dedicated-repo-pr",
+			Conditions: []config.PipelineCondition{
+				{
+					Webhook:    []string{"workflow_run"},
+					Repository: "^mattermost/dedicated$",
+					Type:       "pr",
+					Fork:       false,
+				},
+			},
+		},
+		{
+			Organization: "a",
+			Repository:   "b",
+			Workflow:     "dedicated-repo-branch",
+			Conditions: []config.PipelineCondition{
+				{
+					Webhook:    []string{"workflow_run"},
+					Repository: "^mattermost/dedicated$",
+					Type:       "branch",
+					Fork:       true,
+				},
+			},
+		},
+		{
+			Organization: "a",
+			Repository:   "b",
+			Workflow:     "dedicated-repo-tag",
+			Conditions: []config.PipelineCondition{
+				{
+					Webhook:    []string{"workflow_run"},
+					Repository: "^mattermost/dedicated$",
+					Type:       "tag",
+				},
+			},
+		},
+		{
+			Organization: "a",
+			Repository:   "b",
+			Workflow:     "repo-workflow",
+			Conditions: []config.PipelineCondition{
+				{
+					Webhook:    []string{"workflow_run"},
+					Repository: "^mattermost/workflow$",
+					Type:       "pr",
+					Workflow:   "Build",
+				},
+			},
+		},
+		{
+			Organization: "a",
+			Repository:   "b",
+			Workflow:     "repo-workflow-conclusion",
+			Conditions: []config.PipelineCondition{
+				{
+					Webhook:    []string{"workflow_run"},
+					Repository: "^mattermost/conclusion$",
+					Type:       "pr",
+					Workflow:   "Build",
+					Conclusion: "success",
+				},
+			},
+		},
+		{
+			Organization: "a",
+			Repository:   "b",
+			Workflow:     "repo-workflow-status",
+			Conditions: []config.PipelineCondition{
+				{
+					Webhook:    []string{"workflow_run"},
+					Repository: "^mattermost/status$",
+					Type:       "pr",
+					Workflow:   "Build",
+					Status:     "queued",
+				},
+			},
+		},
+		{
+			Organization: "a",
+			Repository:   "b",
+			Workflow:     "repo-workflow-name",
+			Conditions: []config.PipelineCondition{
+				{
+					Webhook:    []string{"workflow_run"},
+					Repository: "^mattermost/a.*$",
+					Name:       "feat/a.*",
+					Type:       "pr",
+					Workflow:   "Build",
+					Status:     "queued",
+				},
+			},
+		},
+	}
+}
+
+func TestEventContext(t *testing.T) {
+	pipelineConfiguration := createPipelineConfiguration()
+	t.Run("Non Supported Event", func(t *testing.T) {
```

Comment:
There is a nice way in Go for testing which is called [table driven](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests). Some examples https://github.com/mattermost/mattermost-cloud/blob/master/cmd/tools/testwick/wicker/testwicker_test.go#L28

No need for changes, just for the future.